### PR TITLE
Component for applying a raster mask to a raster

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
raster-mask

### CODE REVIEW

**Comment by michaelmattig** [comment]
Some suggestions:
* Preselect two different raster layers, instead of the same two times
* Explain the semantics of the mask. Johannes said he would expect the mask to be multiplied to the raster, meaning values would be scaled if the mask is != 1 for a pixel. Here, everything but a zero means keep the value

**Comment by michaelmattig** [comment]
I tried testing the implementation this way:
image

but I get the following error:
image

**Comment by ChristianBeilschmidt** [comment]
>     * Explain the semantics of the mask. Johannes said he would expect the mask to be multiplied to the raster, meaning values would be scaled if the mask is != 1 for a pixel. Here, everything but a zero means keep the value

The task says
> Raster calculator should result in “nan” at masked pixels (when multiplied by the mask layer)

But how should a multiplication result in NaN if not by accident?

**Comment by ChristianBeilschmidt** [comment]
> I tried testing the implementation this way:
> image
> 
> but I get the following error:
> image

I think I cannot fix this one. Mapping tries to add a NO_DATA value to the values of the unit. But when the unit range covers all values of, for instance, a byte, then it is impossible to add one. And then, we cannot accomplish this task.

File: src/app/operators/dialogs/raster-mask/raster-mask.component.html [hunk 1]
```diff
@@ -1,8 +1,14 @@
 <wave-sidenav-header>Raster Mask</wave-sidenav-header>
 <form [formGroup]="form" (ngSubmit)="add()" fxLayout="column">
+    <wave-dialog-help>
+        The <em>Raster Mask</em> modifies an input raster layer by applying a second mask raster layer.
```
Comment (michaelmattig) [inline src/app/operators/dialogs/raster-mask/raster-mask.component.html]:
```suggestion
        The <em>Raster Mask</em> modifies an input raster layer by applying another raster layer as a mask .
```
there is only one mask

**Pull request merged**